docs/src/literate/computational_homogenization.jl

```diff
@@ -0,0 +1,548 @@
+# - \frac{1}{|\Omega_\Box|} \int_{\Gamma_\Box}\delta \boldsymbol{u} \cdot
+# \boldsymbol{t}\ \mathrm{d}\Gamma = 0 \quad
+# \forall \delta \boldsymbol{u} \in \mathbb{U}_\Box,\quad (1\mathrm{a})\\
+# - \frac{1}{|\Omega_\Box|} \int_{\Gamma_\Box}\delta \boldsymbol{t} \cdot
+# \boldsymbol{u}\ \mathrm{d}\Gamma = - \bar{\boldsymbol{\varepsilon}} :
+# \left[ \frac{1}{|\Omega_\Box|} \int_{\Gamma_\Box}\delta \boldsymbol{t} \otimes
+# [\boldsymbol{x} - \bar{\boldsymbol{x}}]\ \mathrm{d}\Gamma \right]
+# \quad \forall \delta \boldsymbol{t} \in \mathbb{T}_\Box, \quad (1\mathrm{b})
+# ```
+#
+# where ``\boldsymbol{u} = \bar{\boldsymbol{\varepsilon}} \cdot [\boldsymbol{x} -
+# \bar{\boldsymbol{x}}] + \boldsymbol{u}^\mu``, where ``\Omega_\Box`` and ``|\Omega_\Box|``
+# are the domain and volume of the RVE, where ``\Gamma_\Box`` is the boundary, and where
+# ``\mathbb{U}_\Box``, ``\mathbb{T}_\Box`` are set of "sufficiently regular" functions
+# defined on the RVE.
+#
+# This system is not solvable without introducing extra restrictions on ``\mathbb{U}_\Box``,
+# ``\mathbb{T}_\Box``. In this example we will consider the common cases of Dirichlet
+# boundary conditions and (strong) periodic boundary conditions.
+#
+# **Dirichlet boundary conditions**
+#
+# We can introduce the more restrictive sets of ``\mathbb{U}_\Box``:
+#
+# ```math
+# \begin{align*}
+# \mathbb{U}_\Box^\mathrm{D} &:= \left\{\boldsymbol{u} \in \mathbb{U}_\Box|\ \boldsymbol{u}
+# = \bar{\boldsymbol{\varepsilon}} \cdot [\boldsymbol{x} - \bar{\boldsymbol{x}}]
+# \ \mathrm{on}\ \Gamma_\Box\right\},\\
+# \mathbb{U}_\Box^{\mathrm{D},0} &:= \left\{\boldsymbol{u} \in \mathbb{U}_\Box|\ \boldsymbol{u}
+# = \boldsymbol{0}\ \mathrm{on}\ \Gamma_\Box\right\},
+# \end{align*}
+# ```
+#
+# and use these as trial and test sets to obtain a solvable RVE problem pertaining to
+# Dirichlet boundary conditions. Eq. ``(1\mathrm{b})`` is trivially fulfilled, the second
+# term of Eq. ``(1\mathrm{a})`` vanishes, and we are left with the following problem:
+# Find ``\boldsymbol{u} \in \mathbb{U}_\Box^\mathrm{D}`` that solve
+#
+# ```math
+# \frac{1}{|\Omega_\Box|} \int_{\Omega_\Box}\boldsymbol{\varepsilon}[\delta\boldsymbol{u}]
+# : \mathsf{E} : \boldsymbol{\varepsilon}[\boldsymbol{u}]\ \mathrm{d}\Omega = 0
+# \quad \forall \delta \boldsymbol{u} \in \mathbb{U}_\Box^{\mathrm{D},0}.
+# ```
+#
+# Note that, since ``\boldsymbol{u} = \bar{\boldsymbol{\varepsilon}} \cdot [\boldsymbol{x} -
+# \bar{\boldsymbol{x}}] + \boldsymbol{u}^\mu``, this problem is equivalent to solving for
+# ``\boldsymbol{u}^\mu \in \mathbb{U}_\Box^{\mathrm{D},0}``, which is what we will do in
+# the implementation.
+#
+# **Periodic boundary conditions**
+#
+# The RVE problem pertaining to periodic boundary conditions is obtained by restricting
+# ``\boldsymbol{u}^\mu`` to be periodic, and ``\boldsymbol{t}`` anti-periodic across the
+# RVE. Similarly as for Dirichlet boundary conditions, Eq. ``(1\mathrm{b})`` is directly
```

Comment:
"and ``\boldsymbol{t}`` anti-periodic across the RVE"
Is this really true that we enforce, doesn't this just happen to be the case if the structure is periodic?

Reply:
Good question. I believe it is actually just an assumption, but never enforced, which is used to get rid of the second equation.

Reply:
I suppose it must hold true for the integral equation due to equilibrium, but I'm not sure about the point-wise anti-periodicity...

Reply:
It looks like we assume anti-periodicity, and by inserting that assumption in equation 9 it is effectively enforced. Without this assumption, the problem doesn't have sufficient bc because the traction would be unknown for both boundaries. At least that is how it all makes more sense to me now...